File: api/Models/PaymentDetails.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ByzmoApi.Models
{
    public class PaymentDetails
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public string PaymentMethod { get; set; }
        public string PaymentId { get; set; }
        public string PaypalPaymentId { get; set; }
        public string PaypalPayerId { get; set; }
        public string PaypalDebugId { get; set; }
        public string graphQLId { get; set; }
        public int LayAwayId { get; set; }
        public int PreOrderId { get; set; }
        public decimal AmountPaid { get; set; }
        public bool IsTotal { get; set; }
        public bool WithTransactionFee { get; set; }
        public LoyaltyPayment LoyaltyPayment { get; set; }
        public LayawayTransactionFee LayawayTransactionFee { get; set; }
        public PreOrderTransactionFee PreOrderTransactionFee { get; set; }
        public PaypalPayment PaypalPayment { get; set; }

    }
}

File: web/src/app/classes/payment-details.ts
import { LayawayTransactionFee } from './layaway-transaction-fee';
import { LoyaltyPayment } from './loyalty-payment';
import { PayPalPayment } from './paypal-payment';
import { PreOrderTransactionFee } from './pre-order-transactionFee';

export class PaymentDetails {
    id: number;
    orderId: number;
    productId: number;
    paymentMethod: string;
    paymentId: string;
    paypalPaymentId: string;
    paypalPayerId: string;
    paypalDebugId: string;
    graphQLId: string;
    layawayId: number;
    loyaltyPayment: LoyaltyPayment;
    amountPaid: number;
    preOrderId: number;
    isTotal: boolean;
    withTransactionFee: boolean;
    preOrderTransactionFee: PreOrderTransactionFee;
    layawayTransactionFee: LayawayTransactionFee;
    paypalPayment: PayPalPayment;
    pmSourceResourceId: string;
    pmPaymentId: string;

    constructor() {
        this.paypalPaymentId = '';
        this.paypalPayerId = '';
        this.paypalDebugId = '';
        this.paypalPayment = new PayPalPayment();
        this.loyaltyPayment = new LoyaltyPayment();
    }
}

